clients/drcachesim/docs/drcachesim.dox.in

```diff
@@ -703,43 +703,48 @@ An example of thread switches:
+     2214431     1649604:     1249326 ifetch       1 byte(s) @ 0x00007f3402ffa259 55                   push   %rbp
+     2214432     1649604:     1249326 write        8 byte(s) @ 0x00007ffd91e24230 by PC 0x00007f3402ffa259
+     2214433     1649605:     1249326 ifetch       3 byte(s) @ 0x00007f3402ffa25a 48 89 e5             mov    %rsp, %rbp
+     2214434     1649606:     1249326 ifetch       3 byte(s) @ 0x00007f3402ffa25d 89 7d fc             mov    %edi, -0x04(%rbp)
+     2214435     1649606:     1249326 write        4 byte(s) @ 0x00007ffd91e2422c by PC 0x00007f3402ffa25d
+     2214436     1649607:     1249326 ifetch       4 byte(s) @ 0x00007f3402ffa260 48 89 75 f0          mov    %rsi, -0x10(%rbp)
+     2214437     1649607:     1249326 write        8 byte(s) @ 0x00007ffd91e24220 by PC 0x00007f3402ffa260
+     2214438     1649608:     1249326 ifetch       4 byte(s) @ 0x00007f3402ffa264 48 89 55 e8          mov    %rdx, -0x18(%rbp)
+     2214439     1649608:     1249326 write        8 byte(s) @ 0x00007ffd91e24218 by PC 0x00007f3402ffa264
+     2214440     1649609:     1249326 ifetch       4 byte(s) @ 0x00007f3402ffa268 83 7d fc 1a          cmp    -0x04(%rbp), $0x1a
+     2214441     1649609:     1249326 read         4 byte(s) @ 0x00007ffd91e2422c by PC 0x00007f3402ffa268
+     2214442     1649610:     1249326 ifetch       2 byte(s) @ 0x00007f3402ffa26c 75 0f                jnz    $0x00007f3402ffa27d (untaken)
+     2214443     1649611:     1249326 ifetch       6 byte(s) @ 0x00007f3402ffa26e 8b 05 c0 3e 00 00    mov    <rel> 0x00007f3402ffe134, %eax
+     2214444     1649611:     1249326 read         4 byte(s) @ 0x00007f3402ffe134 by PC 0x00007f3402ffa26e
+     2214445     1649612:     1249326 ifetch       3 byte(s) @ 0x00007f3402ffa274 83 c0 01             add    $0x01, %eax
+     2214446     1649613:     1249326 ifetch       6 byte(s) @ 0x00007f3402ffa277 89 05 b7 3e 00 00    mov    %eax, <rel> 0x00007f3402ffe134
+     2214447     1649613:     1249326 write        4 byte(s) @ 0x00007f3402ffe134 by PC 0x00007f3402ffa277
+     2214448     1649614:     1249326 ifetch       1 byte(s) @ 0x00007f3402ffa27d 90                   nop
+     2214449     1649615:     1249326 ifetch       1 byte(s) @ 0x00007f3402ffa27e 5d                   pop    %rbp
+     2214450     1649615:     1249326 read         8 byte(s) @ 0x00007ffd91e24230 by PC 0x00007f3402ffa27e
+     2214451     1649615:     1249326 <marker: indirect branch target 0x7f340665af90>
+     2214452     1649616:     1249326 ifetch       1 byte(s) @ 0x00007f3402ffa27f c3                   ret
+     2214453     1649616:     1249326 read         8 byte(s) @ 0x00007ffd91e24238 by PC 0x00007f3402ffa27f
+     2214454     1649617:     1249326 ifetch       7 byte(s) @ 0x00007f340665af90 48 c7 c0 0f 00 00 00 mov    $0x0000000f, %rax
+     2214455     1649618:     1249326 ifetch       2 byte(s) @ 0x00007f340665af97 0f 05                syscall
+     2214456     1649618:     1249326 <marker: system call 15>
```

Comment:
Hmm, I don't remember considering that: I remember talking about whether it could be inserted at the post-syscall event.  I see your argument for consistency if it's part of the syscall instruction.  But I kind of like it being afterward, if you consider it not to be metadata attached to the syscall instruction but rather a marker at the point where the kernel is doing work.

Reply:
Raise this separately if you would like to discuss further?  Resolving here.